legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/project/ProjectStructure.java

```diff
@@ -446,6 +446,19 @@ else if (requireRevisionId)
         // find out which dependencies we need to update
         boolean updateProjectDependencies = false;
         Set<ProjectDependency> projectDependencies = Sets.mutable.withAll(currentConfig.getProjectDependencies());
+        Set<ProjectDependency> toUpdateProjectDependencies = projectDependencies.stream().filter(dep -> !dep.getProjectId().contains(":")).collect(Collectors.toSet());
+
+        if (toUpdateProjectDependencies.size() > 0)
+        {
+            for (ProjectDependency projectDependency : toUpdateProjectDependencies)
+            {
+                updateProjectDependencies = true;
+                ProjectConfiguration dependencyConfig = getProjectConfiguration(projectFileAccessProvider.getFileAccessContext(projectDependency.getProjectId(),projectDependency.getVersionId()));
+                projectDependencies.remove(projectDependency);
+                projectDependencies.add(ProjectDependency.newProjectDependency(dependencyConfig.getGroupId() + ":" + dependencyConfig.getArtifactId(),projectDependency.getVersionId()));
```

Comment:
You need some null checking here. First, dependencyConfig could be null. Even if it's not, the group or artifact ids (or both) could be null.